Kickstarter-iOS/ViewModels/AppDelegateViewModel.swift

```diff
@@ -282,10 +255,13 @@ AppDelegateViewModelOutputs {
       self.registerForRemoteNotifications = applicationLaunchedWithUser
     }
 
-    self.authorizeForRemoteNotifications = self.notificationAuthorizationStatusProperty.signal
-      .skipNil()
-      .filter { $0.isNotDetermined }
-      .ignoreValues()
+    self.authorizeForRemoteNotifications = Signal.merge(
+      applicationLaunchedWithUser,
+      self.notificationAuthorizationStatusProperty.signal
+        .skipNil()
+        .filter { $0 == .notDetermined }
+        .ignoreValues()
+    )
```

Comment:
May be this is not the best  name for this signal but our `applicationLaunchedWithUser` is fired not only when application launches but also when application enters foreground. One thing I was trying to avoid in my implementation is to call `UNUserNotificationCenter.current().requestAuthorization` every time application about to enter foreground. Not sure if this is important because ios just ignores redundant calls, but thought that this call belongs to the actual  application startup (or user registration).